lib/__tests__/watch.test.ts

```diff
@@ -0,0 +1,191 @@
+import chokidar from 'chokidar';
+import PQueue from 'p-queue';
+
+import { uploadFolder } from '../cms/uploadFolder';
+import { logger } from '../logger';
+import { watch } from '../cms/watch';
+import { MODE } from '../../constants/files';
+
+jest.mock('chokidar');
+jest.mock('axios');
+jest.mock('p-queue');
+jest.mock('../cms/uploadFolder');
+
+describe('watch function', () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let chokidarMock: any;
+  let pQueueAddMock: jest.Mock;
+
+  beforeEach(() => {
+    chokidarMock = {
+      watch: jest.fn().mockReturnThis(),
+      on: jest.fn().mockReturnThis(),
+    };
+    (chokidar.watch as jest.Mock).mockReturnValue(chokidarMock);
+
+    pQueueAddMock = jest.fn();
+
+    // @ts-expect-error test case
+    PQueue.mockImplementation(() => ({
```

Comment:
TypeScript was being overly strict and requiring that I mock out PQueue in its entirety. Opted for this solution instead.